src/engines/v8/v8.zig

```diff
@@ -698,3 +698,59 @@ pub const Inspector = struct {
         return self.session.dispatchProtocolMessage(env.isolate, msg);
     }
 };
+
+// When we return a Zig instance to V8, we wrap it in a v8.Object. That wrapping
+// happens by:
+//  - Assigning our instance to a v8.External (which just holds an *anyopaque)
+//  - Creating a v8.PersistentObject and assigning the external to the
+//    PersistentObject's internalField #0
+//  - Casting the v8.PersistentObject to a v8.Object
+//
+// Now, instead of assigning the instance directly into the v8.External we
+// allocate and assign this ExternalEntry, which allows us to hold the ptr to
+// the Zig instance, as well as meta data that we'll need.
+pub const ExternalEntry = struct {
+    // Ptr to the Zig instance
+    ptr: *anyopaque,
+
+    // When we're asked to describe an object via the Inspector, we _must_ include
+    // the proper subtype (and description) fields in the returned JSON.
+    // V8 will give us a Value and ask us for the subtype. Hence, we store it
+    // here.
+    sub_type: ?[:0]const u8,
```

Comment:
It seems we have a finite list of subtypes: https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#type-RemoteObject
Maybe we could use an enum instead?

Reply:
The list of subtypes is known, from Browser's point of view.  But in zig-js-runtime, just like in V8 itself, it's an arbitrary application-specific value.

I did however change the type to `?[*c]const u8`, which is what we actually need for the V8 interface **and** is 8 bytes smaller. With this change, the `@sizeOf(ExternalEntry)` is the same as-if we were using an enum (assuming padding). Also, this value is, always a string literal, so it's allocation-free.